controllers/applicationsnapshotenvironmentbinding_controller_conditions.go

```diff
@@ -24,30 +24,43 @@ import (
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 func (r *SnapshotEnvironmentBindingReconciler) SetConditionAndUpdateCR(ctx context.Context, req ctrl.Request, appSnapshotEnvBinding *appstudiov1alpha1.SnapshotEnvironmentBinding, createError error) {
 	log := r.Log.WithValues("SnapshotEnvironmentBinding", req.NamespacedName).WithValues("clusterName", req.ClusterName)
 
+	var currentSEB appstudiov1alpha1.SnapshotEnvironmentBinding
+	err := r.Get(ctx, req.NamespacedName, &currentSEB)
+	if err != nil {
+		return
+	}
+
+	patch := client.MergeFrom(currentSEB.DeepCopy())
+	condition := metav1.Condition{}
 	if createError == nil {
-		meta.SetStatusCondition(&appSnapshotEnvBinding.Status.GitOpsRepoConditions, metav1.Condition{
+		condition = metav1.Condition{
 			Type:    "GitOpsResourcesGenerated",
 			Status:  metav1.ConditionTrue,
 			Reason:  "OK",
 			Message: "GitOps repository sync successful",
-		})
+		}
 	} else {
-		meta.SetStatusCondition(&appSnapshotEnvBinding.Status.GitOpsRepoConditions, metav1.Condition{
+		condition = metav1.Condition{
 			Type:    "GitOpsResourcesGenerated",
 			Status:  metav1.ConditionFalse,
 			Reason:  "GenerateError",
 			Message: fmt.Sprintf("GitOps repository sync failed: %v", createError),
-		})
-		logutil.LogAPIResourceChangeEvent(log, appSnapshotEnvBinding.Name, "SnapshotEnvironmentBinding", logutil.ResourceCreate, createError)
+		}
+
 	}
+	meta.SetStatusCondition(&currentSEB.Status.GitOpsRepoConditions, condition)
+	logutil.LogAPIResourceChangeEvent(log, currentSEB.Name, "SnapshotEnvironmentBinding", logutil.ResourceCreate, createError)
+	currentSEB.Status.Components = appSnapshotEnvBinding.Status.Components
```

Comment:
why do we this?
does it get overwritten?

Reply:
we are doing `meta.SetStatusCondition(&currentSEB.Status.GitOpsRepoConditions, condition)`

Reply:
Because we're setting `Status.Components` earlier in the controller on the `appSnapshotEnvBinding` object, but it hasn't been applied server-side, so when we get the "current" object from Kube, it won't have it set. So since setting the status condition on the "current SEB" object, and updating it, I'm also making sure we copy over the `status.Components` status from earlier as well so that it gets applied server-side.

Reply:
nevermind, its `currentSEB.Status.GitOpsRepoConditions` vs `currentSEB.Status.Components`. Sorry Friday afternoon brain fog.

So i guess its because we want to ensure we still have the `appSnapshotEnvBinding.Status.Components` changes we did before calling this function?

Reply:
Yup